### PR TITLE
move golangci-lint to github actions

### PR DESCRIPTION
**Pull request by thaJeztah**
Test run can be seen on https://github.com/thaJeztah/docker/pull/3

### CODE REVIEW

File: .github/workflows/validate.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: src/github.com/docker/docker
+      - name: install dependencies
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt -q update
+          sudo apt -q install \
```
Comment (tianon) [inline .github/workflows/validate.yaml:40]: These are probably printing `WARNING: apt does not have a stable CLI interface. Use with caution in scripts.`, so I'd suggest using `apt-get` here instead:

```suggestion
          sudo apt-get -q update
          sudo apt-get -q install \
```

**Review by tianon — COMMENTED** [review]
Nice! I like it :smile:

File: .github/workflows/validate.yaml (hunk 1 above)
Comment (thaJeztah) [reply .github/workflows/validate.yaml:40]: Damn! Looks like I was caught red-handed doing a copy/pasta from (don't recall where) 😂 I'll update

**Comment by tonistiigi** [comment]
I think the linter should be runnable(same way as it does in CI) without uploading a PR to github. An obvious choice for us would be to run it inside a container/Dockerfile. No objects to invoking that through github actions workflow then.

**Comment by neersighted** [comment]
Closed as this is already done with the new actions CI on master/22.06

**Pull request closed without merging**